eth_abi/base.py

```diff
@@ -144,4 +144,4 @@ def from_type_str(cls, type_str, registry):  # pragma: no cover
         Used by ``ABIRegistry`` to get an appropriate encoder or decoder
         instance for the given type string and type registry.
         """
-        raise NotImplementedError('Must implement `from_type_str`')
+        raise cls()
```

Comment:
This line is confusing to me.

Reply:
Thank you for catching that.  I think I made that change while imagining that it would make some of the examples in the registry documentation more simple but then I just went with slightly more complex examples.  I'll make a PR to fix this right away.

Reply:
PR created for this: #78